### PR TITLE
docs: Add candidate aggregators documentation

### LINKED ISSUE #665: Aggregator tracker
This issue tracks some candidate aggregators that could maybe be added to TorchJD. Feel free to suggest other methods, or to add more information about the methods in the table. If you're interested in implementing one, please tell it so that we can investigate the method in a bit more details (in particular, we need to verify that it can be implemented as an aggregator), and so that we don't work concurrently on the same thing.

Status | Name | Ref | Stateful | Existing implementations | Special Remarks |
|-|-|-|-|-|-|
🟡 In progress (#676) | MoCo (Multi-Objective Gradient Correction) | [Mitigating gradient bias in multi-objective learning](https://openreview.net/pdf?id=dLAYGdKTi2) (ICLR 2023 oral, 95 citations) | Yes (Jacobian EMA) | [LibMTL](https://github.com/median-research-group/LibMTL/blob/main/LibMTL/weighting/MoCo.py) | |
🟣 To Do | GradNorm | [GradNorm: Gradient Normalization for Adaptive Loss Balancing in Deep Multitask Networks](https://proceedings.mlr.press/v80/chen18a/chen18a.pdf) (ICML 2018, 2334 citations) | Yes, trainable state | [LucasBoTang (unofficial)](https://github.com/LucasBoTang/GradNorm/blob/main/gradnorm.py), [lucidrains (unofficial)](https://github.com/lucidrains/gradnorm-pytorch/blob/main/gradnorm_pytorch/gradnorm_pytorch.py), [LibMTL](https://github.com/median-research-group/LibMTL/blob/main/LibMTL/weighting/GradNorm.py)  | See discussion in scalarization channel on discord.
🟠 Too Early | DB-MTL (Dual Balancing Multi-Task Learning) | [Dual-balancing for multi-task learning](https://arxiv.org/pdf/2308.12029) (Neural Networks n°195, 9 citations) | Yes (Jacobian EMA) | [LibMTL (official)](https://github.com/median-research-group/LibMTL/blob/main/LibMTL/weighting/DB_MTL.py) | The paper combines 3 ideas: applying a log to the losses (not something we need TorchJD for), normalizing the gradients (this should be a Normalizer rather than a part of the aggregator) and using the EMA of the gradients. This last idea is also common with some other methods, so maybe we could implement it as a wrapper or something. So in the end I don't think there is anything to implement specifically for DB-MTL. We could make a class combining those ideas, but only after the normalization package is ready. |

### PR DESCRIPTION
**Pull request by KartavyaDikshit**
Closes #665

Implemented automated fix.

### CODE REVIEW

**Comment by KartavyaDikshit** [comment]
Hi maintainers — all tests/checks passed (15 successful), but the only failing check is PR Title Formatter / format-title because this PR does not have a cc: <type> label. Please add exactly one cc: label (suggested: cc: docs) so the title formatter can finish and the PR can be merged. Thanks!

**Comment by PierreQuinton** [comment]
Is this a fully AI generated PR? We would like some human thoughts into PRs.

I think that you would need to read the `CONTRIBUTING.md` file to learn how to contribute. This contribution is not some new aggregators but some new doc strings about aggregator that are not in the library yet.
